test/test5.c

```diff
@@ -1016,55 +1017,70 @@ int test2d(struct Options options)
 	fprintf(xml, "<testcase classname=\"test2d\" name=\"%s\"", testname);
 	global_start_time = start_clock();
 
-	rc = MQTTAsync_create(&c, options.mutual_auth_connection,
-            "test2d", MQTTCLIENT_PERSISTENCE_DEFAULT, NULL);
-	assert("good rc from create", rc == MQTTASYNC_SUCCESS, "rc was %d\n", rc);
-	if (rc != MQTTASYNC_SUCCESS)
-	{
-		MQTTAsync_destroy(&c);
-		goto exit;
-	}
-
-	opts.keepAliveInterval = 20;
-	opts.cleansession = 1;
-	opts.username = "testuser";
-	opts.password = "testpassword";
+        // As reported in https://github.com/eclipse/paho.mqtt.c/issues/190
+        // there is/was some race condition, which caused _sometimes_ that the library failed to detect,
+        // that the connect attempt has already failed.
+        // Therefore we need to test this several times!
+        for (iteration = 0; !failures && (iteration < 20) ; iteration++)
```

Comment:
The number of 20 iterations is not enough. - I have just seen one test run were this tests passed.
But usually the test fails earlier than in the 20th iteration.
It is difficult to determine the requiered number of iterations and also the timeout.